yt_dlp/postprocessor/ffmpeg.py

```diff
@@ -302,6 +302,11 @@ def get_stream_number(self, path, keys, value):
             None)
         return num, len(streams)
 
+    def _fixup_chapters(self, info):
+        last_chapter = traverse_obj(info, ('chapters', -1, {dict})) or {}
+        if 'end_time' in last_chapter and last_chapter['end_time'] is None:
+            info['chapters'][-1]['end_time'] = self._get_real_video_duration(info['filepath'])
```

Comment:
```suggestion
        last_chapter = traverse_obj(info, ('chapters', -1))
        if last_chapter and not last_chapter.get('end_time'):
           last_chapter['end_time'] = self._get_real_video_duration(info['filepath'])
```
We do type checks in IEs because the data is outside our control. We can assume that `chapters` is always a `Iterable[Mapping] | None`